pkg/roachpb/metadata.go

```diff
@@ -650,6 +650,13 @@ func (l Locality) Matches(filter Locality) (bool, Tier) {
 	return true, Tier{}
 }
 
+func (l Locality) NonEmptySlice() []Locality {
```

Comment:
nit: it too me a minute to figure out what this was doing -- I'm not sure why but I find this helper really unintuitive for some reason (maybe since a locality is a slice of tiers itself). what about moving it to pkg/sql/dsp.go  as `func SingleLocalityFilter(Locality) []Locality { <same body> }`?